include/sparrow/array/array_data_factory.hpp

```diff
@@ -39,6 +39,48 @@
 
 namespace sparrow
 {
+
+    // a range where the value type is convertible to bool
+    template<class BoolRange>
+    concept bool_convertible_range = std::ranges::range<BoolRange> &&
+        std::convertible_to<std::ranges::range_value_t<BoolRange>, bool>;
+
+    
+    // a range of nullable values / nullable<T>
+    template<class RangeOfNullables>
+    concept range_of_nullables = std::ranges::range<RangeOfNullables> && is_nullable<std::ranges::range_value_t<RangeOfNullables>>::value;
```

Comment:
Could be moved to mp_utils

Reply:
not sure about that, that would make mpl utils depend on `nullable.hpp`.

We could move `range_of_nullables` to `nullable.hpp` and `bool_convertible_range` to mpl utils